src/linker/Linker/LinkerErrorException.cs

```diff
@@ -1,22 +1,29 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
 
 namespace Mono.Linker
 {
 	public class LinkerErrorException : Exception
 	{
+		public static bool FoundErrors { get; private set; } = false;
```

Comment:
This is weird. We should track whether we reported errors at the place where we actually report them - which is the context/logging. It should not matter if I use `LinkerErrorException` or some other way to produce an error message, all should count the same.

Reply:
With this I'm assuming that the `LinkerErrorException` is the only only way we produce error messages, forcing us to wrap the errors in a `MessageContainer`. I went this way to not have to add a `context.FoundError = true` everywhere we report errors.

Reply:
I understand, but at the same time you now have to explicitly call `FoundErrors` whenever we report non-fatal error. It would be much cleaner to do this at the context level - basically look at the `MessageContainer` and if it's an error - remember that. Nothing else.